include/deal.II/base/mpi.h

```diff
@@ -1102,143 +1104,49 @@ namespace Utilities
            const unsigned int root_process = 0);
 
 
-    template <typename T>
-    struct is_mpi_type
-    {
-      static const bool value = false;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<bool>
-    {
-      static const bool value = true;
-    };
-
-
 
-    template <>
-    struct is_mpi_type<char>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<signed char>
-    {
-      static const bool value = true;
-    };
+    template <bool...>
+    struct check
+    {};
 
 
 
-    template <>
-    struct is_mpi_type<short>
+    template <typename U, typename... T>
+    constexpr auto contains(std::tuple<T...>)
     {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<long int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned char>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned short>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned long int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned long long int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<float>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<double>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<long double>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<std::complex<float>>
-    {
-      static const bool value = true;
-    };
+      return not std::is_same<
+        std::integer_sequence<bool, false, std::is_same<U, T>::value...>,
+        std::integer_sequence<bool, std::is_same<U, T>::value..., false>>::
+        value;
+    }
```

Comment:
Nice! I think we should move this to `include/deal.II/base/template_constraints.h`.

Reply:
Fixed in 8d315c6, is the position correct?